magefile.go

```diff
@@ -1,17 +1,510 @@
+	}
+	log.Println("could not connect to the test cluster")
+	return false
+}
+
+// Ensures that a namespace named "test" exists.
+func EnsureTestNamespace() {
+	mg.Deps(EnsureDeployed)
+	if !namespaceExists(testNamespace) {
+		setupTestNamespace()
+	}
+}
+
+// Ensure ginkgo is installed.
+func EnsureGinkgo() {
+	mgx.Must(pkg.EnsurePackage("github.com/onsi/ginkgo/ginkgo", "", ""))
+}
+
+func setupTestNamespace() {
+	SetupNamespace(testNamespace)
+	EnsureTestSecret()
+	SetupTestCredentialSet()
+}
+
+func namespaceExists(name string) bool {
+	err := kubectl("get", "namespace", name).Must(false).RunS()
+	return err == nil
+}
+
+// Create a namespace, usage: mage SetupNamespace demo.
+// Configures the namespace for use with the operator.
+func SetupNamespace(name string) {
+	mg.Deps(EnsureTestCluster)
+
+	// Only specify the parameter set we have the env vars set
+	// It would be neat if Porter could handle this for us
+	PublishLocalPorterAgent()
+	porter("parameters", "apply", "./hack/params.yaml", "-n=operator").RunV()
+	ps := ""
+	if os.Getenv("PORTER_AGENT_REPOSITORY") != "" && os.Getenv("PORTER_AGENT_VERSION") != "" {
+		ps = "-p=dev-build"
+	}
+
+	porter("invoke", "operator", "--action=configureNamespace", ps, "--param", "namespace="+name, "--param", "porterConfig="+porterConfigFile, "-c", "kind", "-n=operator").
+		CollapseArgs().Must().RunV()
+	kubectl("label", "namespace", name, "--overwrite=true", "porter.sh/devenv=true").Must().RunV()
+
+	setClusterNamespace(name)
+}
+
+// Remove the test cluster and registry.
+func Clean() {
+	mg.Deps(DeleteTestCluster, StopDockerRegistry)
+	os.RemoveAll("bin")
+}
+
+func CleanLastTestRun() {
+	os.RemoveAll(".cnab")
+}
+
+func pwd() string {
+	wd, _ := os.Getwd()
+	return wd
+}
+
+func EnsureTestSecret() {
+	if !testSecretExists() {
+		setupTestSecret()
+	}
+}
+func testSecretExists() bool {
+	err := kubectl("-n", testNamespace, "get", "secrets", "password").Must(false).RunS()
+	return err == nil
+}
+func setupTestSecret() {
+	kubectl("create", "secret", "generic", "password", "--from-literal=credential=test", "-n", testNamespace).Must(false).Run()
+}
+
+func SetupTestCredentialSet() {
+	kubectl("-n", testNamespace, "apply", "-f", "./tests/integration/operator/testdata/agent_action_create_password_creds.yaml").Must(false).Run()
+}
+
+func TestInstallation() {
+	kubectl("-n", testNamespace, "create", "-f", "./tests/integration/operator/testdata/porter-test-me.yaml").Must(false).Run()
+}
+
+func kubectl(args ...string) shx.PreparedCommand {
+	kubeconfig := fmt.Sprintf("KUBECONFIG=%s", os.Getenv("KUBECONFIG"))
+	return must.Command("kubectl", args...).Env(kubeconfig)
+}
+
+// Run porter using the local storage, not the in-cluster storage
+func porter(args ...string) shx.PreparedCommand {
+	return must.Command("bin/porter").Args(args...).
+		Env("PORTER_DEFAULT_STORAGE=",
+			"PORTER_DEFAULT_STORAGE_PLUGIN=mongodb-docker",
+			fmt.Sprintf("PORTER_HOME=%s", porterHome))
+}
+
+func PublishLocalPorterAgent() {
```

Comment:
Is this the function that people should use instead of BuildLocalPorterAgent? I'm referring to the docs in the contributing guide, where we explain which targets to use when.